resources/js/functions.js

```diff
@@ -323,6 +323,48 @@ $(document).ready(function () {
   }
 });
 
+$(document).ready(function () {
+  // set up automatic switching of code carousel
+  $(".code-carousel").each(function () {
+    var carousel = this;
+    var inputs = [];
+    $(carousel).children("input.code-carousel_control").each(function () {
+      inputs.push(this);
+    });
+
+    // if there is more than one section, set up automatic switching
+    if (inputs.length > 1) {
+
+      // at every timeout, switch to the next section if the page is visible
+      const Ticker = (timeout) => {
```

Comment:
This abstraction seems to obscure the intent more than it clarifies it. Consider removing that lambda and instead directly execute its body.

Reply:
removed it